app/controllers/reports_controller.rb

```diff
@@ -37,11 +37,11 @@ def weekly_report
     @title = "Weekly Report"
     date = Date.parse(params[:date]) rescue 1.day.ago
 
-    @sprint = Sprint.find_by_date!(date)
-    @checked_out_by ||= Hash[SprintTask
-      .where(sprint_id: @sprint.id, task_id: @sprint.tasks.pluck(:id))
-      .includes(:checked_out_by)
-      .map { |task| [task.task_id, task.checked_out_by] }]
+    @sprint = Sprint.from_date(date)
+    # @checked_out_by ||= Hash[SprintTask
+    #   .where(sprint_id: @sprint.id, task_id: @sprint.tasks.pluck(:id))
+    #   .includes(:checked_out_by)
+    #   .map { |task| [task.task_id, task.checked_out_by] }]
```

Comment:
Maybe, if we're not using Sprints anymore, we drop that concept and just call this `Week`? Sprint, with this change, is just a wrapper for what a week is, given a date.

(And drop the comments 🙂)

Reply:
Great idea! Done ✅ 😁